react/src/views/Box/components/HistoryEntries.tsx

```diff
@@ -0,0 +1,60 @@
+import { ListItem, ListIcon, List, Stack, Flex } from "@chakra-ui/react";
+import { IconType } from "react-icons";
+import { MdCheckCircle, MdSettings, MdHistory } from "react-icons/md";
+import { HistoryEntry } from "types/generated/graphql";
+
+interface IHistoryEntriesProps {
+  data: HistoryEntry[];
+  total: number | undefined;
+}
+function HistoryEntries({ data, total }: IHistoryEntriesProps) {
+  // sort history entries in the descending order
+  // this can be optimize in the BE by changing the defualt query order or adding sort option
+  const historyData = [...data].sort(
+    (a, b) => new Date(b.changeDate).getTime() - new Date(a.changeDate).getTime(),
+  );
+
+  // show different icons for changed / created records also default history for the rest
+  const getHistoryIcon = (changes: string): IconType => {
+    if (changes.includes("created")) {
+      return MdSettings;
+    }
+    if (changes.includes("changed")) {
```

Comment:
If changes include both "created" and "changed", this block overrides the first icon, is that on purpose? Otherwise `else if` probably makes sense

Reply:
@pylipp it was on purpose but we can change later if needed